src/network/networkd-state-file.c

```diff
@@ -606,6 +608,17 @@ int link_save(Link *link) {
 
                 /************************************************************/
 
+                if (link->dhcp_lease && link->network->dhcp_use_captive_portal) {
+                        r = sd_dhcp_lease_get_captive_portal(link->dhcp_lease, &dhcp_captive_portal);
+                        if (r < 0 && r != -ENODATA)
+                                return r;
+                }
+
+                if (dhcp_captive_portal)
+                        fprintf(f, "CAPTIVE_PORTAL=%s\n", dhcp_captive_portal);
```

Comment:
should probably use cescape() to ensure any embedded newlines in the URLs don't confuse things